### DISCUSSION TITLE
Transcribing uncommon words with whisper

### DISCUSSION THREAD

**Discussion by OthmanePreure** [openai/whisper#2169] [Q&A]
I'm transcribing railway related audio files, however the model gets most of the Time the name of the train stations wrong
I tried fine tuning whisper with audio files (2 hours) but the results wasn't good enough. then i tried with prompt but the size of the context is limited to 244 token which in note sufficient regarding the important number of stations.

Any tips to solve this issue (maybe using some methods with post processing)

**Comment by glangford** [comment]
Post-processing with soundex or metaphone or some other form of fuzzy matching? eg. checking Whisper generated spellings against a dictionary of properly spelled station names


### Phonetics based Fuzzy string matching algorithms - A dive in Soundex & Metaphone
https://medium.com/data-science-in-your-pocket/phonetics-based-fuzzy-string-matching-algorithms-8399aea04718

**Comment by rahulbansal16** [comment]
@OthmanePreure Did you try the phonetics-based matching?
